src/cogent3/core/sequence.py

```diff
@@ -1024,6 +1024,10 @@ def __add__(self, other):
         else:
             other_seq = other
         new_seq = self.__class__(self._seq + other_seq)
```

Comment:
This is python 3.8+ syntax, so the tests will fail on python 3.7.
Also, `self._seq` is a string so will not have a `.name` attribute.

Replace the code with the following if expression and then create the instance
```python
name = self.name if self.name == other.name else None
new_seq = self.__class__(self._seq + other_seq, name=name)
```